netcfgbu/config_model.py

```diff
@@ -35,121 +39,210 @@
 
 
 class NoExtraBaseModel(BaseModel):
-    class Config:
-        extra = "forbid"
+    """A base model class that forbids extra fields in derived models."""
 
+    model_config = ConfigDict(extra="forbid")
 
-class EnvExpand(str):
-    """
-    When a string value contains a reference to an environment variable, use
-    this type to expand the contents of the variable using os.path.expandvars.
 
-    For example like:
-        password = "$MY_PASSWORD"
-        foo_password = "${MY_PASSWORD}_foo"
+def expand_env_str(env_string: str) -> str:
+    """Expand environment variables in a string.
 
-    will be expanded, given MY_PASSWORD is set to 'boo!' in the environment:
-        password -> "boo!"
-        foo_password -> "boo!_foo"
-    """
+    Args:
+        env_string: The string containing environment variable references.
 
-    @classmethod
-    def __get_validators__(cls):
-        yield cls.validate
+    Returns:
+        str: The string with expanded environment variables.
 
-    @classmethod
-    def validate(cls, v):
-        if found_vars := list(filter(len, chain.from_iterable(_var_re.findall(v)))):
-            for var in found_vars:
-                if (var_val := os.getenv(var)) is None:
-                    raise ValueError(f'Environment variable "{var}" missing.')
+    Raises:
+        EnvironmentError: If an environment variable is missing or empty.
+    """
+    if found_variables := list(filter(len, chain.from_iterable(_var_re.findall(env_string)))):
+        for var in found_variables:
+            if (var_val := os.getenv(var)) is None:
+                raise OSError(f"Environment variable '{var}' missing.")
 
-                if not len(var_val):
-                    raise ValueError(f'Environment variable "{var}" empty.')
+            if not var_val:
+                raise OSError(f"Environment variable '{var}' empty.")
```

Comment:
**Suggestion:** The `expand_env_str` function should handle the case where the environment variable is found but is empty, by either using a default value or by continuing without raising an exception, depending on the desired behavior. [enhancement, importance: 5]
```suggestion
if not var_val:
    var_val = "default_value"  # Provide a default value or handle accordingly
```